Client/Frontend/Browser/BackForwardListViewController.swift

```diff
@@ -4,56 +4,203 @@
+        setupTableView()
+        loadSites(backForwardList)
+    }
+    
+    func setupTableView() {
+        tableView.separatorStyle = .None
+        tableView.dataSource = self
+        tableView.delegate = self
+        tableView.alwaysBounceVertical = false
+        tableView.registerClass(BackForwardTableViewCell.self, forCellReuseIdentifier: BackForwardListCellIdentifier)
+        
+        tableView.backgroundColor = isPrivate ? BackForwardViewUX.BackgroundColorPrivate:BackForwardViewUX.BackgroundColor
+        let blurEffect = UIBlurEffect(style: isPrivate ? .Dark : .ExtraLight)
+        let blurEffectView = UIVisualEffectView(effect: blurEffect)
+        tableView.backgroundView = blurEffectView
+    }
+    
+    func loadSites(backForwardList: WKBackForwardList) {
+        let sql = profile.favicons as! SQLiteHistory
+        var urls: [String] = [String]()
+        
+        for page in backForwardList.forwardList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Forward))
         }
-
-        listTableView.snp_makeConstraints { make in
-            make.top.equalTo(toolbar.snp_bottom)
+        
+        var currentRow = 0
+        if let currentPage = backForwardList.currentItem {
+            currentRow = listData.count
+            urls.append(currentPage.URL.absoluteString)
+            listData.append((currentPage, .Current))
+        }
+        for page in backForwardList.backList.reverse() {
+            urls.append(page.URL.absoluteString)
+            listData.append((page, .Backward))
+        }
+        
+        listData = listData.filter { !(($0.item.title ?? "").isEmpty)}
+        
+        sql.getSitesForURLs(urls).uponQueue(dispatch_get_main_queue()) { result in
+            if let cursor = result.successValue {
+                for cursorSite in cursor {
+                    if let site = cursorSite, let url = site?.url {
+                        self.sites[url] = site
+                    }
+                }
+                self.tableView.reloadData()
+            }
+        }
+        
+        if currentRow > 1 {
+            scrollTableViewToIndex(currentRow)
+        }
+    }
+    
+    func scrollTableViewToIndex(index: Int) {
+        let moveToIndexPath = NSIndexPath(forRow: index-2, inSection: 0)
+        self.tableView.reloadRowsAtIndexPaths([moveToIndexPath], withRowAnimation: .None)
+        self.tableView.scrollToRowAtIndexPath(moveToIndexPath, atScrollPosition: UITableViewScrollPosition.Middle, animated: false)
+    }
+    
+    override func didRotateFromInterfaceOrientation(fromInterfaceOrientation: UIInterfaceOrientation) {
+        self.resizeHeight()
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.addSubview(tableView)
+        tableView.snp_makeConstraints { make in
+            make.height.equalTo(0)
             make.left.right.bottom.equalTo(self.view)
         }
+        self.view.layoutIfNeeded()
```

Comment:
I think this layoutIfNeeded is needed for the animation because it's creating the constraints and then animating the height in the same call.  Otherwise it seems to animate from the origin (0,0,0,0) to open, it seems to ignore the "snp_makeConstraints".